src/tree-sitter-language-mode.js

```diff
@@ -637,13 +637,14 @@ class TreeSitterLanguageMode {
 }
 
 class LanguageLayer {
-  constructor(languageMode, grammar, contentChildTypes) {
+  constructor(languageMode, grammar, depth, isOpaque) {
     this.languageMode = languageMode;
     this.grammar = grammar;
     this.tree = null;
     this.currentParsePromise = null;
     this.patchSinceCurrentParseStarted = null;
-    this.contentChildTypes = contentChildTypes;
+    this.depth = depth;
+    this.isOpaque = isOpaque;
```

Comment:
I added a boolean flag on `LanguageLayer` that I'm calling `isOpaque`, which indicates whether or not the parent layer's token should *show through* this layer or not. It basically matches the value of `includeChildren`. So by default, when `includeChildren` is false, the parent layer's highlighting does show. But for injections with `includeChildren: true`, the parent layer is covered up.